### PR TITLE
[no-release-notes] add persist interface

### PR DESCRIPTION
**Pull request by JCOR11599**
there is now a persist interface that is used to determine how to save mysql db data

### CODE REVIEW

File: sql/plan/create_role.go [hunk 1]
```diff
@@ -106,6 +106,12 @@ func (n *CreateRole) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error)
 	if !ok {
 		return nil, sql.ErrDatabaseNotFound.New("mysql")
 	}
+
+	// Check if you can even persist in the first place
+	if !mysqlDb.CanPersist() {
+		return nil, fmt.Errorf("no privilege file specified, to persist users/grants run with --privilege-file=<file_path>")
```
Comment (zachmu) [inline sql/plan/create_role.go:112]: This error message belongs in dolt, not here

File: sql/mysql_db/mysql_db.go [hunk 2]
```diff
@@ -30,9 +30,26 @@ import (
 	"github.com/dolthub/go-mysql-server/sql/mysql_db/serial"
 )
 
-// PersistCallback represents the callback that will be called when the Grant Tables have been updated and need to be
-// persisted.
-type PersistCallback func(ctx *sql.Context, data []byte) error
+// MySQLDbPersistence is used to determine the behavior of how certain tables in MySQLDb will be persisted.
+type MySQLDbPersistence interface {
+	CanPersist() bool
```
Comment (zachmu) [inline sql/mysql_db/mysql_db.go:35]: Make this return an error instead, call it `ValidateCanPersist()`

Then integrators can return their own error message if they can't persist the data

File: sql/plan/create_user.go [hunk 3]
```diff
@@ -100,6 +100,10 @@ func (n *CreateUser) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error)
 	if !ok {
 		return nil, sql.ErrDatabaseNotFound.New("mysql")
 	}
+	// Check if you can even persist in the first place
+	if !mysqlDb.CanPersist() {
```
Comment (zachmu) [inline sql/plan/create_user.go:104]: Forgetting `GRANT` statements, right?

**Review by zachmu — APPROVED** [review]
Mostly good, just one small tweak

**Review by Hydrocharged — APPROVED** [review]
LGTM

**Pull request merged**